plugins/file-open-save-new/core/src/main/java/org/pentaho/di/plugins/fileopensave/providers/vfs/VFSFileProvider.java

```diff
@@ -88,6 +126,26 @@ public class VFSFileProvider extends BaseFileProvider<VFSFile> {
     return TYPE;
   }
 
+  /**
+   * Determines is a <code>filePath</code> is a VFS file
+   * @param filePath
+   * @return true if VFS file, false otherwise.
+   */
+  public boolean isSupported( String filePath ) {
+    if ( filePath == null ) {
+      return false;
+    }
+    boolean ret = false;
+    try {
+      URI uriFilePath = new URI( filePath );
+      String testScheme = uriFilePath.getScheme() + "://";
+      ret = uriFilePath.getScheme() != null && testScheme.matches( ConnectionFileSystem.DOMAIN_ROOT );
```

Comment:
isn't the "nullness" of `uriFilePath.getScheme()` accounted for in the `test.Scheme.matches` in that a null scheme will result in a testScheme of "://", which will not match `ConnectionFileSystem.DOMAIN_ROOT` of `"[\\w]+://";` given that it expects one or more preceding words?

Not an issue, honestly probably cleaner to short-circuit the logic on `getScheme()` being null, but, I had a thought about it, so, figured I'd share :D